models/DataObject/ClassDefinition/Helper/PathFormatterResolver.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @category   Pimcore
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model\DataObject\ClassDefinition\Helper;
+
+use Pimcore\Logger;
+use Pimcore\Model\DataObject\ClassDefinition\PathFormatterInterface;
+
+class PathFormatterResolver
+{
+    public static $formatterCache = [];
+
+    /**
+     * @param $formatterClass
+     *
+     * @return PathFormatterInterface
+     */
+    public static function resolvePathFormatter($formatterClass): ?PathFormatterInterface
+    {
+        if ($formatterClass) {
+            $formatter = null;
+
+            if (isset(self::$formatterCache[$formatterClass])) {
+                return self::$formatterCache[$formatterClass];
+            }
+            if (substr($formatterClass, 0, 1) == '@') {
+                $serviceName = substr($formatterClass, 1);
+                try {
+                    $formatter = \Pimcore::getKernel()->getContainer()->get($serviceName);
```

Comment:
formatter not resolved when using DI in Formatter Class config

Reply:
why not? whats your test scenario?

Reply:
Sorry, there was some issue with defining service at my end. maybe good idea is to add simple example for service configuration when using DI method.

Reply:
it is and was always with `@SERVICENAME`...

Reply:
**Doc updates:**
I mean provide service example something like this https://github.com/pimcore/pimcore/blob/master/doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/08_Dynamic_Select_Types.md 

Also update preferred way to use path formatter with DI and add deprecation message for old one

**Class definition:**
change lable to
![image](https://user-images.githubusercontent.com/5137917/56677253-640f7780-66c0-11e9-8a34-86bb182c1f82.png)

Reply:
Deprecation Message: https://github.com/pimcore/pimcore/pull/3399/files#diff-385d3cb9b74d2e9378fedd615fd53648R797

Example: https://github.com/pimcore/pimcore/pull/3399/files#diff-9d97d6e06d55f5e9f21a271e3e61d282R17

label: https://github.com/pimcore/pimcore/pull/3399/files#diff-8b61221e64319cf4539f459c80217173R513